pkg/apis/core/validation/validation.go

```diff
@@ -264,6 +264,33 @@ func ValidateRuntimeClassName(name string, fldPath *field.Path) field.ErrorList
 	return allErrs
 }
 
+// ValidateOverhead can be used to check whether the given Overhead is valid.
+func ValidateOverhead(overhead *core.ResourceRequirements, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	limPath := fldPath.Child("limits")
+	reqPath := fldPath.Child("requests")
+
+	//Overhead validation is a superset of Resource Requirements
+	allErrs = append(allErrs, ValidateResourceRequirements(overhead, fldPath)...)
+
+	//If an overhead for a resource is being defined, both limits and request should be set
```

Comment:
What does this mean for best-effort pods? Does it even make sense to have separate requests and limits for overhead?

I would like the QoS to be preserved, independent of RuntimeClass (assuming the runtime supports all QoS - I'm not sure if BestEffort makes sense for Kata). I can't remember if the KEP covered this?

Reply:
> What does this mean for best-effort pods? Does it even make sense to have separate requests and limits for overhead?

I think it makes sense to have seperate still ('burstable').  As this PR is written right now, I was under the mindset that leaving 'limits' unset was a bad idea.  Ie, even for a burstable Overhead, there should be a limit defined. I no longer think this makes sense; leaving it able to scale unbound on host could be useful, and it doesn't make sense to change the existing pattern on how ResourceRequirements are used today.

> I would like the QoS to be preserved, independent of RuntimeClass (assuming the runtime supports all QoS - I'm not sure if BestEffort makes sense for Kata). I can't remember if the KEP covered this?

For a workload, unbound burstable (limits not set) are not well support with Kata (ie, additional vCPUs are added based on a limits value provided).

Reply:
TL;DR  -- I'll update validation/default logic to match what is done today for core.ResourceRequirements .